artman/pipelines/batch_generation.py

```diff
@@ -92,25 +92,28 @@ def _get_api_kwarg_dicts(
         api_config = _load_artman_config(api_config_yaml,
                                          language,
                                          local_paths)
-        if not api_config.get('enable_batch_generation', True):
-            continue
         api_kwargs = lang_config.copy()
         api_kwargs.update(api_config)
         yield api_kwargs
 
 
-def _get_artman_config_filenames(api_config_patterns, batch_apis):
+def _get_artman_config_filenames(api_config_patterns, batch_apis, exclude_apis):
     # We consider two separate cases: when batch_apis='*', and when batch_apis
     # is a list. In the first case, we find all files which match the patterns
     # listed in api_config_patterns. In the second case, for each api listed in
     # batch_apis, we try to find a matching file in EXACTLY ONE of the
     # api_config_patterns.
-    config_filenames= []
+    # In both cases, entries in exclude_apis override entries in batch_apis,
+    # meaning that if an API appears in both lists, it will not be included.
+    exclude_api_file_set = frozenset(exclude_apis)
+    config_filenames = []
     if batch_apis == '*':
         for pattern in api_config_patterns:
             glob_pattern = config_util.replace_vars(pattern,
-                                                   {'API_SHORT_NAME': '*'})
-            config_filenames += sorted(glob.glob(glob_pattern))
+                                                    {'API_SHORT_NAME': '*'})
```

Comment:
I'm not sure what this is referring to - to you mean that `{'API_SHORT_NAME': '*'})` is offset by one space? That is intentional, to correct the alignment.